quinn-proto/src/connection.rs

```diff
@@ -906,12 +908,8 @@ impl Connection {
             }
             trace!(self.log, "read {} TLS bytes", n);
             if let Err(e) = self.tls.read_handshake(&buf[..n]) {
-                debug!(self.log, "TLS error: {}", e);
-                return Err(if let Some(alert) = self.tls.alert() {
-                    TransportError::crypto(alert)
-                } else {
-                    TransportError::PROTOCOL_VIOLATION("TLS error")
-                });
+                debug!(self.log, "reading handshake bytes resulted in {}", e);
```

Comment:
`TransportError`s are already logged in `handle_packet`, and we're now encoding all information into them, so this is redundant. Should be able to do away with the `if let` entirely.

Reply:
Fixed.